gatk-launch

```diff
@@ -105,36 +125,89 @@ def main(args):
     except CalledProcessError as e:
         sys.exit(e.returncode)
 
-
-
 def getSparkSubmit():
     sparkhome = os.environ.get("SPARK_HOME")
     if sparkhome is not None:
         return sparkhome +"/bin/spark-submit"
     else:
         return "spark-submit"
 
+def getLocalGatkRunCommand():
+    shadowJarFromEnv = getJarFromEnv(GATK_SHADOW_JAR_ENV_VARIABLE)
+    if shadowJarFromEnv is not None:
+        return ["java", "-jar", shadowJarFromEnv]
+
+    wrapperScript = getGatkWrapperScript(throwIfNotFound=False)
+    if wrapperScript is not None:
+        return [wrapperScript]
 
-def getSparkJar():
-    sparkproperty = os.environ.get('GATK_SPARK_JAR')
-    if not sparkproperty is None:
-        if not os.path.exists(sparkproperty):
-            raise GATKLaunchException("GATK_SPARK_JAR was set to: " + sparkproperty + " but it doesn't exist, please fix your environment")
+    return ["java", "-jar", getShadowJar()]  # will throw if shadow jar not found
+
+def getGatkWrapperScript(throwIfNotFound=True):
+    if not os.path.exists(GATK_RUN_SCRIPT):
+        if throwIfNotFound:
+            raise GATKLaunchException("Missing GATK wrapper script: " + GATK_RUN_SCRIPT + "\nTo generate the wrapper run:\n\n    " + script + "/gradlew installDist")
         else:
-            return sparkproperty
-
-    if not os.path.exists(BIN_PATH):
-        raise GATKLaunchException("No spark jar was found, please build one by running\n\n    " + script + "/gradlew sparkJar")
-    files = os.listdir(BIN_PATH)
-    sparkjars = [f for f in files if "spark.jar" in f]
-    if len(sparkjars) is 0:
-        raise GATKLaunchException("No spark jar was found, please build one by running\n\n    " + script + "/gradlew sparkJar\n"
-                             "or\n"
-                             "    export GATK_SPARK_JAR=<path_to_spark_jar>")
-    else:
-        newest = max(sparkjars, key=lambda x: os.stat(BIN_PATH + "/" + x).st_mtime)
-        return BIN_PATH+ "/" + newest
+            return None
 
+    sys.stderr.write("Using GATK wrapper script " + GATK_RUN_SCRIPT)
+    return GATK_RUN_SCRIPT
+
+def getShadowJar(throwIfNotFound=True):
+    shadowJar = findJar("shadowJar.jar", envVariableOverride=GATK_SHADOW_JAR_ENV_VARIABLE)
```

Comment:
It's `shadowJar.jar` because that's the suffix that gradle uses for our shadow jar -- I could attempt to re-configure `build.gradle` to use different names, though I think if we did that we'd have to rename `gradle shadowJar` to something else to be consistent.